### ISSUE TITLE
Deletion of files in e2e folders no longer possible as of v3.24

### ISSUE THREAD

**Issue by tigernero79** [nextcloud/android#11387]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

Nextcloud 3.24.0 does not allow deletion of files in e2e folders

does not exit option eliminated among those proposed by the 3 dots menu

### Expected behaviour

Nextcloud 3.24.0 does not allow deletion of files in e2e folders

does not exit option eliminated among those proposed by the 3 dots menu

### Actual behaviour

Nextcloud 3.24.0 does not allow deletion of files in e2e folders

does not exit option eliminated among those proposed by the 3 dots menu

### Android version

13

### Device brand and model

Pixel 6 pro

### Stock or custom OS?

Stock

### Nextcloud android app version

3.24.0

### Nextcloud server version

25.0.3

### Using a reverse proxy?

No

**Comment by AlvaroBrey** [comment]
cc @tobiasKaminsky

**Comment by tigernero79** [comment]
News?

**Comment by tigernero79** [comment]
you posted an update to your app saying it fixed the delete menu problem in e2e folders.  which is not true to date it is still not possible to delete an encrypted file in the e2e folder via the android application, because you have updated the app saying to solve it when it is not true?

**Comment by tigernero79** [comment]
@tobiasKaminsky Are you by any chance aware of this problem is there a fix?

**Comment by tigernero79** [comment]
@tobiasKaminsky Hi Tobias, I noticed that it's been more than a month that the android version of nextcloud app does not receive updates, nor is this important bug mentioned e2e in the impossibility of android to delete files or folders e2e, it works with ios version.

but the project is stopped? few collaborators?

also predictions of future release 3.25.1 scheduled for 11/7/23 no hint what's up?

**Comment by tigernero79** [comment]
any news to resolve this very important bug for e2e folders?

**Comment by joshtrichards** [comment]
@tigernero79 Does this report mean that your other report (#11155) can be closed?

**Comment by tigernero79** [comment]
it means that the error of not being able to delete files or folders if you imitate e2ee unfortunately still remains even with nextcloud 28.0.1 and app 1.14, but even worse, application 1.14 with nextcloud 28.0 and 28.01 is no longer even recognized.  in practice, even being the desktop/android and iOS application, it is not active.  but what happens to us?  e2ee as managed is rubbish.  both bugs must be kept active

**Comment by joshtrichards** [comment]
My point was that the situation you reported in #11155 would prevent the usage of E2EE in Android. I'm trying to determine how you got as far as attempting to delete E2EE folders - if your Android client isn't recognizing E2EE as even being enabled (as you reported in #11155).

**Comment by tigernero79** [comment]
The problem that the delete file or folder item does not appear in the drop-down menu on Android was there even before the second bug appeared. if you notice the error it dates back a long time and persists. the second bug is current and does not allow recognition that e2e is active on all platforms. but do you communicate with each other? with Nextcloud 28 and the latest e2e 1.14 app on the server side, although e2e is enabled in both the desktop PC and Android and iOS versions, it appears not to be active. as for the bug where there is no option to delete e2e files or folders in Android, this was the case until the application completely messed up everything. clear?

**Comment by mawumag** [comment]
This behavior is due to commit 1d62ee7 , in particular the snippet (there are analogous ones for Move/Copy and Rename):
```
if (files.isEmpty() || synchronizing || containsLockedFile()
    || containsEncryptedFolder() || containsEncryptedFile()) {
    toHide.add(R.id.action_remove_file);
}
```
Reverting this commit, I can add/remove files and subfolders inside the main encrypted folder with no issue. However I notice that I cannot remove the root encrypted folder without making it empty and removing encryption first (I get a "server unavailable" error if I do not follow these steps). Maybe that is why the checks above were introduced in the first place?

@tobiasKaminsky , would it be possible to limit those checks to the root encrypted folder _only_ and re-allow the Delete, Move/Copy, and Rename? If it's nontrivial I'd be happy to volunteer to try to find a solution.

**Comment by mawumag** [comment]
It is easy to filter out the "Delete" option only for the root encrypted directory, using for example something like

```
    private boolean isEncryptedRoot() {
        return isEncryptedFolder() && !hasEncryptedParent();
    }
```
I've tested the above and it works appropriately, both for files and (nested) folders.

---

However, the Move/Copy and Rename operations are not yet implemented for encrypted files (see [app/src/main/java/com/owncloud/android/operations](https://github.com/nextcloud/android/tree/master/app/src/main/java/com/owncloud/android/operations)). The server throws the error `Forbidden Write access to end-to-end encrypted folder requires token - no token sent`.

Is this something that's already in your pipeline? Otherwise I'd be happy to give it a go (the Desktop client has these capabilities, as well as recursive deletion for non-empty root encrypted folders).

**Comment by joshtrichards** [comment]
The main work in progress I'm aware of is sharing related. Perhaps we can get @tobiasKaminsky to pop in - he'd be the authoritative source in this particular area. Would you mind posting your Delete option implementation as a PR?

**Comment by tigernero79** [comment]
I think Tobias ran away on holiday.  a significant open bug for some time.  quoted to tobias multiple times but never responded.  the android version is a lot behind the ios version to manage working

**Comment by joshtrichards** [comment]
@tigernero79:

> I think Tobias ran away on holiday. a significant open bug for some time. quoted to tobias multiple times but never responded.

Well Tobias doesn't work for you and this is open source so that just means they've got other priorities. :-)

There have been numerous E2EE related commits made since your original report (by Tobias and others):
https://github.com/nextcloud/android/pulls?q=is%3Apr+in%3Atitle+%28e2ee+OR+e2e%29+is%3Aclosed+

(And that's not counting the ones elsewhere such as in the `android-library` repository)

> you have released a new update describing in the properties the possibility of deleting files in the e2e folders, but this is not true, the delete item still does not exit
> 3.24.2 (March 31, 2023)
>
>    E2E file drop

I think you may have misunderstood the above. "File drop" has nothing to do with delete.

**Comment by tigernero79** [comment]
I know the project is open source, I don't know what you mean by delete files, that's not what I mean.  but I'll tell you what I mean if something works and then it doesn't work anymore I call it a bug.  if I have files in the e2ee and android app folder there is no longer an entry that allows you to delete them when there was before, I call it a bug. If I turn to the maintainers of the Android NextCloud GitHub and I haven't received any replies since February, I call it a lack of seriousness. .  I know Tobias is busy, I don't blame anyone.  but after a year of the problem not being solved, how would you define all this?  it being understood that if someone writes here it's because he cares.  Having said that, good job.  3.24.2 was released saying in the resolution notes on the problem of deleting e2ee files and folders, which is not true. To date, the Android application does not allow this.

**Comment by joshtrichards** [comment]
> Having said that, good job. 3.24.2 was released saying in the resolution notes on the problem of deleting e2ee files and folders, which is not true. To date, the Android application does not allow this.

No I was trying to point out that the release notes are about the "file drop" feature in Nextcloud[1]. File drop is about public/anonymous uploads. They are not related to deletions. ;-)

> the "Rename" function which exists in the IOS version is missing, I wonder is the function to rename files in the E2E folders missing or is it also a bug?

Can you open a separate Issue about that since this one is for deletion?

[1] https://docs.nextcloud.com/server/latest/user_manual/en/files/file_drop.html

**Comment by joshtrichards** [comment]
Fixed in #11922 
Also thanks to @mawumag for the initial targeted fix.

**Comment by gcvl** [comment]
The menu item is there but it's not working at all: "Eliminazione non riuscita" (deletion failed).
Also, I erased the file manually from the desktop PC but the app does not sync and I can still access the deleted file.
